test/libsolidity/AnalysisFramework.h

```diff
@@ -153,6 +153,15 @@ do \
 } \
 while(0)
 
+#define CHECK_SUCCESS_OR_WARNING(text, substring) \
+do \
+{ \
+	auto sourceAndError = parseAnalyseAndReturnError((text), true); \
+	if (!sourceAndError.second.empty()) \
+		CHECK_WARNING(text, substring); \
```

Comment:
Wait... maybe we should avoid parsing and analyzing twice, just realizing that this does...